app/src/main/java/org/oppia/app/player/state/itemviewmodel/FractionInteractionViewModel.kt

```diff
@@ -1,20 +1,55 @@
 package org.oppia.app.player.state.itemviewmodel
 
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.Log
+import androidx.databinding.Bindable
+import org.oppia.app.BR
 import org.oppia.app.model.InteractionObject
 import org.oppia.app.parser.StringToFractionParser
 import org.oppia.app.player.state.answerhandling.InteractionAnswerHandler
+import org.oppia.app.topic.FractionParsingErrors
 import org.oppia.domain.util.toAnswerString
 
 class FractionInteractionViewModel(
   existingAnswer: InteractionObject?, val isReadOnly: Boolean
-): StateItemViewModel(), InteractionAnswerHandler {
+) : StateItemViewModel(), InteractionAnswerHandler {
   var answerText: CharSequence = existingAnswer?.toAnswerString() ?: ""
 
   override fun getPendingAnswer(): InteractionObject {
     val interactionObjectBuilder = InteractionObject.newBuilder()
-    if (answerText.isNotEmpty()) {
+    if (answerText.isNotEmpty() && StringToFractionParser().fromRawInputString(answerText.toString()) == FractionParsingErrors.VALID) {
```

Comment:
I don't think we need to check correctness here. We should design the code such that we don't even call getPendingAnswer if the current answer text is invalid.

Reply:
Reverted